algorithms/gr-dfsrh/javascript/dfs.js

```diff
@@ -0,0 +1,41 @@
+
+const dfs = (graph, startNode) => {
+    // Create an array / fake "stack" for storing next visiting nodes
+    let stack = [startNode];
+    let visited = [];
+
+    while (stack.length) {
+        // Take first node from stack to visit
+        node = stack.shift();
+
+        // Check whether the node is not visited 
+        if (visited.indexOf(node) === -1) {
+            visited.push(node);
+            console.log(`visited ${node}`);
+            
+            // Find direct-connected node and put to stack if it's not visited
+            graph.edges.filter(edge => edge.start == node && !visited.includes(edge.end)).forEach(item => {
```

Comment:
Codacy found an issue: ['node' is not defined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396906)